src/lib.rs

```diff
@@ -857,10 +857,39 @@ impl BumpFeeTxBuilder {
     }
 }
 
-fn generate_mnemonic(word_count: WordCount) -> Result<String, BdkError> {
-    let mnemonic: GeneratedKey<_, BareCtx> =
-        Mnemonic::generate((word_count, Language::English)).unwrap();
-    Ok(mnemonic.to_string())
+struct Mnemonic {
+    mnemonic_mutex: Mutex<BdkMnemonic>,
```

Comment:
I think we may not need to wrap this inner `BdkMnemonic` in a Mutex. But I still need to test it out to be sure.
```suggestion
    inner: BdkMnemonic,
```

Reply:
Okay, so it should be like this instead
```
struct Mnemonic {
    inner: BdkMnemonic
}

impl Mnemonic {
    fn new(...) -> Arc<Self> {...}
    fn from_string(...) -> Result<Arc<Self>, BdkError> {...}
    fn from_entropy(...) -> Result<Arc<Self>, BdkError> {...}
    fn as_string(...) -> String {...}
}
```
Right?

Reply:
Thanks, yes but the API should look the same, uniffi-rs is smart enough to know that since it's marked as a constructor in the UDL file that it will add an `Arc` to the result in the glue code it makes.
```rust
struct Mnemonic {
    inner: BdkMnemonic
}

impl Mnemonic {
    fn new(...) -> Self {...}
    fn from_string(...) -> Result<Self, BdkError> {...}
    fn from_entropy(...) -> Result<Self, BdkError> {...}
    fn as_string(...) -> String {...}
}
```

Reply:
Updated `Mnemonic` struct according to suggestion. Resolving conversation.